jetbrains-core/tst/software/aws/toolkits/jetbrains/services/lambda/sam/SamExecutableTest.kt

```diff
@@ -0,0 +1,170 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.lambda.sam
+
+import com.intellij.execution.configurations.GeneralCommandLine
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TemporaryFolder
+import software.aws.toolkits.jetbrains.services.lambda.deploy.CreateCapabilities
+
+class SamExecutableTest {
+    @Rule
+    @JvmField
+    val tempFolder = TemporaryFolder()
+
+    @Test
+    fun `sam build command is correct`() {
+        val templatePath = tempFolder.newFile("template.yaml").toPath()
+        val buildDir = tempFolder.newFolder("build").toPath()
+        val cmd = GeneralCommandLine("sam").samBuildCommand(
+            environmentVariables = mapOf("Foo" to "Bar"),
+            templatePath = templatePath,
+            buildDir = buildDir,
+            useContainer = false
+        )
+
+        assertThat(cmd.commandLineString).isEqualTo(
+            listOf(
+                "sam",
+                "build",
+                "--template",
+                "$templatePath",
+                "--build-dir",
+                "$buildDir"
+            ).joinToString(separator = " ")
+        )
+
+        assertThat(cmd.workDirectory).isEqualTo(tempFolder.root)
+
+        assertThat(cmd.environment).containsEntry("Foo", "Bar")
```

Comment:
Should also check the common environment variables for the credentials

Reply:
We dont pass creds at this layer, its already in the GeneralCommandLine before this is invoked.